core/modules/node/node.block.inc

```diff
@@ -0,0 +1,163 @@
+<?php
+
+/**
+ *  NodeBlock  extends Block
+ *  
+ *  This class allows us to display a particular node in a block.
+ */
+class NodeBlock extends Block {
+  /**
+   *  Sets title text on draggable block panel in Layout builder.
+   */
+  function getAdminTitle() {
+    return t('Content block');
+  }
+
+  /**
+   * Return a preview for this block.
+   */
+  function getAdminPreview() {
+    return t('Displays content of node ID: @nid', array('@nid' => $this->settings['nid']));
+  }
+
+  /**
+   *  Sets block subject on block view.
+   */
+  function getTitle() {
+    $title = NULL;
+    $node = node_load($this->settings['nid']);
+    if ($this->settings['title_display'] === LAYOUT_TITLE_CUSTOM && $this->settings['title']) {
+      $title = check_plain($this->settings['title']);
+    }
+    elseif ($this->settings['title_display'] === LAYOUT_TITLE_DEFAULT) {
+      $title = $node->title;
+    }
+
+    if ($title && $this->settings['link_node_title']) {
+      $title = l($title, 'node/' . $node->nid);
+    }
```

Comment:
Needs an `else { check_plain($title); }`

`l()` will check_plain content for you by default.